internal/charmstore/store.go

```diff
@@ -570,9 +570,13 @@ func (s *Store) findEntityInChannel(url *charm.URL, ch params.Channel, fields ma
 	}
 	var entityURL *charm.URL
 	if url.Series == "" {
-		for _, u := range baseEntity.ChannelEntities[ch] {
-			if entityURL == nil || seriesScore[u.Series] > seriesScore[entityURL.Series] {
+		var entitySeries string
+		for s, u := range baseEntity.ChannelEntities[ch] {
```

Comment:
// Choose the URL with the highest series score. If there are
// several multi-series URLs, choose the one associated
// with the highest scoring series.

?